.github/workflows/cncf-conformance-test.yaml

```diff
@@ -0,0 +1,45 @@
+name: CNCF Conformance Nightly Latest/Edge Tests
+
+on:
+  schedule:
+    - cron: '0 0 * * 1' # Runs every monday midnight
+  pull_request:
+permissions:
+  contents: read
+
+jobs:
+  test-cncf-conformance:
+    name: CNCF Conformance ${{ matrix.os }} ${{ matrix.arch }} ${{ matrix.releases }}
+    strategy:
+      matrix:
+        os: ["ubuntu:24.04"]
+        arch: ["amd64"]
+        releases: ["latest/edge/moonray"] # TODO add latest/edge after upgrade Cilium to 1.17
+    runs-on: ${{ matrix.arch == 'arm64' && 'Ubuntu_ARM64_4C_16G_01' || 'ubuntu-20.04' }}
+
+    steps:
+      - name: Checking out repo
+        uses: actions/checkout@v4
+
+      - name: Install lxd
+        run: |
+          sudo snap refresh lxd --channel 5.21/stable
+          sudo lxd init --auto
+          sudo usermod --append --groups lxd $USER
+          sg lxd -c 'lxc version'
+          echo "os_dash=${{ matrix.os }}" | sed 's/:/-/g' >> $GITHUB_ENV
+
+      - name: download ${{ matrix.releases }} k8s snap
+        run: |
+          snap download k8s --channel=${{ matrix.releases }} --basename k8s
+
+      - name: Run end to end tests on ${{ matrix.arch }} ${{ matrix.os }}
+        run: |
+          sg lxd -c 'build-scripts/hack/sonobuoy.sh ${{ matrix.arch }} ${{ matrix.os }}'
```

Comment:
Could we please have the snap release as an argument to this script and download the snap inside of the script? It'll be easier to read than `/repo/k8s.snap`. (I plan on using your test in k8s-dqlite as well)